assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts

```diff
@@ -0,0 +1,146 @@
+import { prepareEntitiesForUpdate, ticketsWithNewQuantity } from './';
+import { nodes as allDates } from '@edtrServices/apollo/queries/datetimes/test/data';
+import { nodes as allTickets } from '@edtrServices/apollo/queries/tickets/test/data';
+import { relationalData as existingData } from '@edtrServices/context/TestContext/data';
+import { assocPath, pathOr } from 'ramda';
+
+describe('TAM:ticketsWithNewQuantity', () => {
+	it('returns an empty object when existing relational data is empty', () => {
+		const ticketsToUpdate = prepareEntitiesForUpdate({
+			entity: 'tickets',
+			existingData: {},
+			newData: {},
+			relation: 'datetimes',
+		});
+
+		const ticketsWithChangedQuantity = ticketsWithNewQuantity({
+			allDates,
+			allTickets,
+			existingData,
+			ticketsToUpdate,
+		});
+
+		expect(ticketsWithChangedQuantity).toEqual({});
+	});
+
+	it('returns an empty object when new relational data is empty', () => {
+		const ticketsToUpdate = prepareEntitiesForUpdate({
+			entity: 'tickets',
+			existingData,
+			newData: {},
+			relation: 'datetimes',
+		});
+
+		const ticketsWithChangedQuantity = ticketsWithNewQuantity({
+			allDates,
+			allTickets,
+			existingData,
+			ticketsToUpdate,
+		});
+
+		expect(ticketsWithChangedQuantity).toEqual({});
+	});
+
+	it('returns an empty object when assignment is only removed', () => {
+		// lets modify existing data to create new one
+		const path = ['tickets', allTickets[0].id, 'datetimes'];
+		// ticket at allTickets[0] is related to two dates
+		// lets assign the ticket to only one date and remove others
```

Comment:
but if ticket is assigned to one date shouldn't this return this only relation?

Reply:
nvm. it's checking `ticketsWithNewQuantity`

Reply:
Yes, and the logic is same